### PR TITLE
Cleanup ruby code in privileges UI.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Before adding another option to the dropdown (the `levelbuilder` option), I'm trying to make the existing code be more readable and better conform to our style guidelines. Though I should extract this ruby from the HAML to a ruby helper, I am punting on this for now.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
CC @bcjordan

File: pegasus/sites.v3/code.org/public/private/privileges/index.haml [hunk 1]
```diff
@@ -14,50 +14,39 @@
   messages = []
 
   def grant_permission_to_email(email, permission_name)
-    exit = false
```
Comment (bcjordan) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml:17]: 🎉

**Comment by ashercodeorg** [comment]
PTAL, made another change preventing requeries to the DB.

**Comment by ashercodeorg** [comment]
Note that all this code assumes there is at most one account per email address. In theory, this is true. In practice, it is not.

File: pegasus/sites.v3/code.org/public/private/privileges/index.haml [hunk 2]
```diff
@@ -14,50 +14,39 @@
...
-        if !exit
-          created_at = DateTime.now
-          PERMISSIONS.insert(user_id:user[:id], permission:permission_name, created_at:created_at, updated_at:created_at)
-          message = "#{email} granted '#{permission_name}'"
-        end
-        return message
-      end
-    else
-      return "#{email} is not a user"
+    if !PERMISSIONS.where(user_id: user[:id], permission: permission_name).empty?
```
Comment (bcjordan) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml]: `if PERMISSIONS.where(user_id: user[:id], permission: permission_name).any?` might work to un-double-negative

Comment (bcjordan) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml]: (or `.present?` if we have ActiveSupport in this context)

**Comment by bcjordan** [comment]
LGTM!

File: pegasus/sites.v3/code.org/public/private/privileges/index.haml [hunk 3]
```diff
@@ -14,50 +14,39 @@
   messages = []
 
   def grant_permission_to_email(email, permission_name)
-    exit = false
-
-    if user = USERS.where(email:email).first
-      #user = USERS.where(email:email).first
+    user = USERS.where(email: email).first
+    if !user
+      return "#{email} is not a user"
```
Comment (bcjordan) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml:19]: Nit (know this is just moved code): admin might think this means they should input a username instead. Could maybe be `No user with email #{email}`

File: pegasus/sites.v3/code.org/public/private/privileges/index.haml (hunk 2 above)
Comment (ashercodeorg) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml]: Done.

File: pegasus/sites.v3/code.org/public/private/privileges/index.haml (hunk 3 above)
Comment (ashercodeorg) [inline pegasus/sites.v3/code.org/public/private/privileges/index.haml:19]: Going to punt on changing the text, given that the view clearly indicates to input email addresses.

**Pull request merged**